integration-tests/controller-scripts/resultsProcessor.js

```diff
@@ -1,25 +1,28 @@
 const fs = require("fs");
+const path = require('path');
 const xml2js = require("xml2js");
 const { mergeFiles } = require('junit-report-merger');
-const processResult = async (directoryPath, originalFileName, outputFileName, runtimeId) => {
+
+const processResult = (reportsFolder, fileName) => {
+    // sample fileName; fluent-bit_2.0.8_debian-bullseye_amd64.deb.xml
     const parser = new xml2js.Parser();
-    fs.readFile(`${directoryPath}/${originalFileName}`, async (err, data) => {
-        parser.parseString(data, async function (err, result) {
-            console.log(runtimeId);
-            const runtimeProperties = runtimeId.split("-");
+    const runtimeId = path.parse(fileName).name;
+    const runtimeProperties = runtimeId.split('_');
+    const data = fs.readFileSync(`${reportsFolder}/${fileName}`);
+        parser.parseString(data, function (err, result) {
```

Comment:
Is this syntax convertable to `parser.parseString(data).then(...)`? Forgive my rusted JS knowledge here

Reply:
Nope, its just how the api is written. I mean, it does not return a promise, its a void method.